### PR TITLE
Feature/49065 view void transfer for aa aatf pcs pbs int ext

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteDataMap.cs [hunk 1]
```diff
@@ -24,12 +24,12 @@ public ListOfEvidenceNoteDataMap Map(ListOfNotesMap source)
...
             {
                 foreach (var note in source.ListOfNotes)
-                { 
+                {
                     var evidenceNoteData = mapper.Map<EvidenceNoteWithCriteriaMap, EvidenceNoteData>(new EvidenceNoteWithCriteriaMap(note)
                     {
                         CategoryFilter = source.CategoryFilter,
                         IncludeTonnage = source.IncludeTonnage,
-                        IncludeHistory = false
+                        IncludeHistory = note.Status == Domain.Evidence.NoteStatus.Void
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteDataMap.cs:32]: Set to false here. We dont want to include history when this is the notes to be displayed in the tables.

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs [hunk 2]
```diff
@@ -91,6 +91,9 @@ public ViewTransferNoteViewModel Map(ViewTransferNoteViewModelMapTransfer source
                     null),
                 TransferredByAddress = transferredByFormattedAddress,
                 Summary = GenerateNotesModel(source),
+
+                EvidenceNoteHistoryData = null,  //TODO: when transfer history appears at top of view page
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs:95]: Can be removed a transfer note wont have the history its the original evidence note that has the history. George to update the wireframe.

File: src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs [hunk 3]
```diff
@@ -219,7 +219,8 @@ public async Task<ActionResult> TransferredEvidence(Guid pcsId, Guid evidenceNot
                     noteData, TempData[ViewDataConstant.TransferEvidenceNoteDisplayNotification])
                 {
                     RedirectTab = redirectTab,
-                    SystemDateTime = currentDateTime
+                    SystemDateTime = currentDateTime, 
+                    IncludeHistory = true
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs:223]: Can be removed a transfer note wont have the history its the original evidence note that has the history. George to update the wireframe.

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMapTransfer.cs [hunk 4]
```diff
@@ -20,6 +20,8 @@ public class ViewTransferNoteViewModelMapTransfer
 
         public DateTime SystemDateTime { get; set; }
 
+        public bool IncludeHistory { get; internal set; }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMapTransfer.cs:23]: Can be removed a transfer note wont have the history its the original evidence note that has the history. George to update the wireframe.

File: src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/_ViewTransferEvidenceNotePartial.cshtml [hunk 5]
```diff
@@ -5,6 +5,19 @@
...
 @model EA.Weee.Web.ViewModels.Shared.ViewTransferNoteViewModel
 
+<!-- screen 6.09a: View Void Transfer History -->
+@if (Model.DisplayEvidenceNoteHistoryData)
+{
+    <div class="govuk-width-container govuk-!-padding-left-0">
+        @(this.WeeeGds().CreateProgressiveDisclosure("Transfer history",
+                        @Html.Partial("~/Views/Shared/_EvidenceTransferProgressiveDisclosurePartial.cshtml").ToString(),
+                        "Hidden content",
+                        "Progressive disclosure",
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/_ViewTransferEvidenceNotePartial.cshtml:15]: Can be removed a transfer note wont have the history its the original evidence note that has the history. George to update the wireframe.

File: src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs [hunk 6]
```diff
@@ -81,7 +81,7 @@ public ViewEvidenceNoteViewModel Map(ViewEvidenceNoteMapTransfer source)
                 AatfApprovalNumber = source.EvidenceNoteData.AatfData.ApprovalNumber,
                 DisplayEditButton = (source.EvidenceNoteData.Status == NoteStatus.Draft || source.EvidenceNoteData.Status == NoteStatus.Returned) && source.EvidenceNoteData.AatfData.CanCreateEditEvidence,
                 RedirectTab = source.RedirectTab,
-                EvidenceNoteHistoryData = mapper.Map<IList<EvidenceNoteHistoryViewModel>>(source.EvidenceNoteData.EvidenceNoteHistoryData),
+                EvidenceNoteHistoryData = mapper.Map<IList<EvidenceNoteHistoryViewModel>>(source.EvidenceNoteData.EvidenceNoteHistoryData)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs:84]: Can be removed a transfer note wont have the history its the original evidence note that has the history. George to update the wireframe.

File: src/EA.Weee.Web/ViewModels/Shared/EvidenceNoteRowViewModel.cs [hunk 7]
```diff
@@ -77,6 +77,8 @@ public string SchemeViewRouteName
                         return SchemeTransferEvidenceRedirect.ViewRejectedTransferEvidenceRouteName;
                     case NoteStatus.Returned:
                         return SchemeTransferEvidenceRedirect.ViewReturnedTransferEvidenceRouteName;
+                    case NoteStatus.Void:
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/ViewModels/Shared/EvidenceNoteRowViewModel.cs:80]: Update the related unit test for this new status and route name.

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteDataMap.cs (hunk 1 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteDataMap.cs:32]: done

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs (hunk 2 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMap.cs:95]: done

File: src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs (hunk 3 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs:223]: done

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMapTransfer.cs (hunk 4 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ViewTransferNoteViewModelMapTransfer.cs:23]: done

File: src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/_ViewTransferEvidenceNotePartial.cshtml (hunk 5 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Views/TransferEvidence/_ViewTransferEvidenceNotePartial.cshtml:15]: done

File: src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs (hunk 6 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs:84]: are you sure about this one ?

File: src/EA.Weee.Web/ViewModels/Shared/EvidenceNoteRowViewModel.cs (hunk 7 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/ViewModels/Shared/EvidenceNoteRowViewModel.cs:80]: done

File: src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs:84]: Ah good spot. It looked like a new line was added but just a comma removed, ignore me. on this one.

File: src/EA.Weee.Web.Tests.Unit/ViewModels/EvidenceNoteRowViewModelTests.cs [hunk 8]
```diff
@@ -127,16 +127,16 @@ public void AatfViewRouteName_GivenRejectedNoteStatus_RejectedViewShouldBeReturn
...
+        public void AatfViewRouteName_GivenVoidNoteStatus_VoidedViewShouldBeReturned()
         {
             //arrange
             model.Status = NoteStatus.Void;
 
             //act
-            var exception = Record.Exception(() => model.AatfViewRouteName);
+            var result = model.AatfViewRouteName;
 
             //assert
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/ViewModels/EvidenceNoteRowViewModelTests.cs:138]: @GuidoWorkDev  Just noticed, is this correct? The AATF AatfViewRouteName returns a scheme named route SchemeTransferEvidenceRedirect.ViewVoidedTransferEvidenceRouteName?

File: src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs (hunk 6 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs:84]: done

File: src/EA.Weee.Web.Tests.Unit/ViewModels/EvidenceNoteRowViewModelTests.cs (hunk 8 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web.Tests.Unit/ViewModels/EvidenceNoteRowViewModelTests.cs:138]: added some code to resolve AATF view void note

**Pull request merged**